File: tutorials/30-days-of-code/30-linked-list.py
# Tutorials > 30 Days of Code > Day 15: Linked List
# Complete the body of a function that adds a new node to the tail of a Linked List.
#
# https://www.hackerrank.com/challenges/30-linked-list/problem
#

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
class Solution:
    def display(self,head):
        current = head
        while current:
            print(current.data,end=' ')
            current = current.next
# (skeliton_head) ----------------------------------------------------------------------

    def insert(self,head,data):
        #Complete this method

        if head is None:
            return Node(data)
        node = head
        while node.next is not None:
            node = node.next
        node.next = Node(data)
        return head

# (skeliton_tail) ----------------------------------------------------------------------
mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)
mylist.display(head);

File: tutorials/30-days-of-code/30-linked-list.cpp
// Tutorials > 30 Days of Code > Day 15: Linked List
// Complete the body of a function that adds a new node to the tail of a Linked List.
//
// https://www.hackerrank.com/challenges/30-linked-list/problem
//

#include <iostream>
#include <cstddef>
using namespace std;
class Node
{
    public:
        int data;
        Node *next;
        Node(int d){
            data=d;
            next=NULL;
        }
};
class Solution{
    public:
// (skeliton_head) ----------------------------------------------------------------------

      Node* insert(Node *head,int data)
      {
          //Complete this method
          if (head == NULL)
          {
            return new Node(data);
          }
          Node *node = head;
          while (node->next)
          {
              node = node->next;
          }
          node->next = new Node(data);
          return head;
      }

// (skeliton_tail) ----------------------------------------------------------------------
void display(Node *head)
      {
          Node *start=head;
          while(start)
          {
              cout<<start->data<<" ";
              start=start->next;
          }
      }
};
int main()
{
	Node* head=NULL;
  	Solution mylist;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        head=mylist.insert(head,data);
    }
	mylist.display(head);

}
